asmcli/components/ca/managed-cas.sh

```diff
@@ -75,6 +75,59 @@ x_enable_workload_certificate_on_membership() {
      -d "${ENABLEFEATURE}" "https://${GKEHUB_API}/v1alpha/projects/${FLEET_ID}/locations/global/features/workloadcertificate?update_mask=membership_specs"
 }
 
+x_wait_for_gke_hub_api_enablement() {
+  info "Wait 3 minutes for GKE HUB API enablement to propagate to the systems ..."
+  sleep 180
+}
+
+x_wait_for_enabling_workload_certificates() {
+  local GKEHUB_API; GKEHUB_API="$1"
+  local FLEET_ID; FLEET_ID="$2"
+
+  info "Waiting for the workload certificates enablement. This may take up to 20 minutes ..."
+  # Check the status of workload certificate feature
+  local ENABLED; ENABLED=0
+  for i in {1..2}; do
+    local OUT; OUT="$(curl -H "X-Goog-User-Project: ${FLEET_ID}" \
+      -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+      "https://${GKEHUB_API}/v1alpha/projects/${FLEET_ID}/locations/global/features/workloadcertificate" 2>/dev/null | \
+      grep -i "certificateManagement.*ENABLED")"
+    if [[ -z "$OUT" ]]; then
+      info "Workload certificate management is enabled."
+      ENABLED=1
+      break
+    else
+      info "Workload certificate management is not enabled, try again ..."
+      sleep 60
+    fi
+  done
+  if [[ "${ENABLED}" -eq 0 ]]; then
+    warn "The workload certificate management is not enabled. The workload certificates may not work until the management is enabled."
+  fi
+
+  # Enabling workload certificate feature on a cluster will result in
+  # the cluster restarting and the cluster becoming unreachable for minutes.
+  # The cluster restart may happen at a random time after the enablement.
+  # Therefore, wait 15 minutes before checking the connection to k8s is recovered.
+  sleep 900
+  # Check the connection to k8s is recovered.
+  local REACHABLE; REACHABLE=0
+  for i in {1..3}; do
+    local OUT; OUT=$(kubectl version -o json 2>/dev/null | jq .serverVersion.major -r 2>/dev/null)
```

Comment:
@williamaronli - wonder if we should instead wait to check if node daemon set is installed, instead of checking if the kube api server is up?

Reply:
Once the cluster is restarted successfully and the cluster should already is installed with zatar

Right now we use the command: ```kubectl get ds -n kube-system gke-spiffe-node-agent``` could we use this in this script?

Reply:
@lei-tang - was thinking this is better since this actually tests if the cluster is correctly configured/ready for managed certificates

Reply:
Done.